### PR TITLE
Updated JS+Rails section titles & description

### PR DESCRIPTION
**Pull request by Saranoya**
Changed JS+Rails section titles and description to accommodate the changes I made to the curriculum in response to [Issue #11148](https://github.com/TheOdinProject/curriculum/issues/11148) - see [this pull request](https://github.com/TheOdinProject/curriculum/pull/16844) for details.

### CODE REVIEW

File: db/seeds/06_javascript_course_seeds.rb [hunk 1]
```diff
@@ -474,18 +474,18 @@
...
-  title: "JavaScript and Rails",
-  title_url: "JavaScript and Rails".parameterize,
+  title: "JavaScript and the Backend",
+  title_url: "JavaScript and the Backend".parameterize,
   course_id: course.id,
   position: section_position,
-  description: "A real web app needs a back end in order to persist its data and do sensitive operations. Here you'll learn how to use ajax to send data requests to your Rails back end. You'll also get a chance to work with external APIs like Google Maps, which is something you'll probably find yourself doing quite frequently as you build your own projects in the future. Note: if you've opted to skip the Ruby/Rails portion of the curriculum to do (only) full stack JS, you can safely skip this unit, too."
+  description: "A real web app needs a back end in order to persist its data and do sensitive operations. Here you'll learn how to use ajax to send data requests to your Rails back end, and/or how to outsource your backend to a Backend-as-a-Service company like Firebase. You'll also get a chance to work with external APIs, which is something you'll probably find yourself doing quite frequently as you build your own projects in the future."
```
Comment (mindovermiles262) [inline db/seeds/06_javascript_course_seeds.rb:481]: I think we can clean this up:

```
A real web app needs a back end in order to persist its data and do sensitive operations. Here you'll learn how to use ajax to accomplish this. You'll ...
```

File: db/seeds/06_javascript_course_seeds.rb [hunk 2]
```diff
@@ -474,18 +474,18 @@
...
 )
 
 lesson_position += 1
 create_or_update_lesson(
-  title: "Using Ruby on Rails For Your Back End",
-  title_url: "Using Ruby on Rails For Your Back End".parameterize,
-  description: "You've got experience working with APIs, now it's time to treat your Rails app like one.",
+  title: "Using Ruby on Rails or BaaS For Your Back End",
```
Comment (mindovermiles262) [inline db/seeds/06_javascript_course_seeds.rb:486]: I think "Using a back end" would be a better title here

File: db/seeds/06_javascript_course_seeds.rb (hunk 1 above)
Comment (Saranoya) [reply db/seeds/06_javascript_course_seeds.rb:481]: Well ... the people who opt to use Firebase won't be learning AJAX.

File: db/seeds/06_javascript_course_seeds.rb (hunk 2 above)
Comment (Saranoya) [reply db/seeds/06_javascript_course_seeds.rb:486]: Done.

**Comment by Saranoya** [comment]
> Can you please update the `README.md` with the changes as well?

Do you mean [this one](https://github.com/TheOdinProject/theodinproject/blob/master/README.md)? I don't see anything there that warrants a change on account of this update ...

**Pull request merged**